### ISSUE TITLE
Support for Mistral-Small-Instruct-2409

### ISSUE THREAD

**Issue by AniketRajpoot** [TUDB-Labs/MoE-PEFT#35]
Hi, thank you for your amazing work. 

I have been running experiments with Llama 3.1 base and it seems to perform really well for some tasks. I was wondering if the existing implementation will support Mistral-Small-Instruct-2409. 

Furthermore, are there any plans to support newer LLMs as well as multi-gpu training for larger models ? 

Thank you!

**Comment by mikecovlee** [comment]
It looks like the [Mistral-Small-Instruct-2409 config](https://huggingface.co/mistralai/Mistral-Small-Instruct-2409/blob/main/config.json) is using the existing MistralForCausalLM model adapter, which is currently supported by MoE-PEFT. You can download and use it directly. If you encounter any issues, please feel free to leave a message here. We’ll be happy to help! :)

**Comment by AniketRajpoot** [comment]
Thanks for sharing this @mikecovlee, I trained Mistral-Small-Instruct-2409 and Mistral-7B-Instruct-v0.1 and after few steps the loss seems to climb up and then it never goes down, is there something I am doing wrong ? 

I am attaching both mixlora config and logs for the loss for both models, just one FYI from my end I modified prompter.py to generate prompt using tokenizer's prompt template as I am training MixLoRA on top of instruct variants, thanks in advance and please let me know if you need any further information!  

Mistral 2409 config : 
```
{
    "cutoff_len": 512,
    "save_step": 1000,
    "train_lora_candidate_num": 2,
    "train_lora_simultaneously_num": 2,
    "train_strategy": "optim",
    "lora": [
        {
            "name": "mixlora_mistral_sarcasm_reddit_0",
            "task_name": "casual",
            "optim": "adamw",
            "scheduler_type": "constant",
            "warmup_steps": 0,
            "lr": 0.0002,
            "batch_size": 8,
            "micro_batch_size": 4,
            "evaluate_batch_size": 1,
            "num_epochs": 2,
            "r": 16,
            "lora_alpha": 32,
            "lora_dropout": 0.05,
            "target_modules": {
                "q_proj": true,
                "k_proj": true,
                "v_proj": true,
                "o_proj": true,
                "gate_proj": true,
                "down_proj": true,
                "up_proj": true
            },
            "routing_strategy": "mixlora",
            "num_experts": 3,
            "top_k": 2,
            "group_by_length": false,
            "data": "data/besstie/Sarcasm/Reddit/merged.json",
            "prompt": "mistral"
        }
    ]
}
```

Mistral 7B config :

```
{
    "cutoff_len": 512,
    "save_step": 1000,
    "train_lora_candidate_num": 2,
    "train_lora_simultaneously_num": 2,
    "train_strategy": "optim",
    "lora": [
        {
            "name": "mixlora_mistral_sarcasm_reddit_0",
            "task_name": "casual",
            "optim": "adamw",
            "scheduler_type": "constant",
            "warmup_steps": 0,
            "lr": 0.0002,
            "batch_size": 8,
            "micro_batch_size": 4,
            "evaluate_batch_size": 1,
            "num_epochs": 2,
            "r": 16,
            "lora_alpha": 32,
            "lora_dropout": 0.05,
            "target_modules": {
                "q_proj": true,
                "k_proj": true,
                "v_proj": true,
                "o_proj": true,
                "gate_proj": true,
                "down_proj": true,
                "up_proj": true
            },
            "routing_strategy": "mixlora",
            "num_experts": 3,
            "top_k": 2,
            "group_by_length": false,
            "data": "data/besstie/Sarcasm/Reddit/merged.json",
            "prompt": "mistral"
        }
    ]
}
```

Train Logs for Mistral 2409 : 
[mistral_2409.txt](https://github.com/user-attachments/files/23572269/mistral_train_sarcasm.txt)
[mistral_7b_v0.1.txt](https://github.com/user-attachments/files/23573142/mistral_7b_v0.1_sarcasm.txt)

Edit : I also tried training on base variant using alpaca template and the loss is increasing after ~500 steps / 6480. Also I was able to successfully train Llama 3.1 and Qwen 2.5 with same config and dataset.

**Comment by mikecovlee** [comment]
We have noticed your problem and will research on this problem. Thanks for using MoE-PEFT!

**Comment by mikecovlee** [comment]
BTW I think you can adjust your settings of `"num_experts": 3` to larger number such as 6 or 8. Also your learning rate is a little bit to high.

**Comment by AniketRajpoot** [comment]
Thanks @mikecovlee, I have already tried reducing the learning rate `2e-5`. I initially set `"num_experts": 3` to mirror the number of dialects in our dataset, but I’m happy to experiment with increasing it to 6 or 8.

Please let me know your thoughts, and if there are any updates on this.
